src/PIL/Image.py

```diff
@@ -1531,6 +1531,23 @@ def getpalette(self, rawmode="RGB"):
             rawmode = mode
         return list(self.im.getpalette(mode, rawmode))
 
+    def has_transparency_data(self):
+        """
+        Determine if an image has transparency data, whether in the form of an
+        alpha channel, a palette with an alpha channel, or a "transparency" key
+        in the info dictionary.
+
+        Note the image might still appear solid, if all of the values shown
+        within are opaque.
+
+        :returns: A boolean.
+        """
+        return (
+            self.mode in ("LA", "La", "PA", "RGBA", "RGBa")
+            or (self.mode == "P" and self.palette.mode == "RGBA")
```

Comment:
Correct way to check P mode from my application:

```python
        if self.mode == 'P':
            self.load()
            return self.im.getpalettemode().endswith('A')
```

I can't remember all details, I had problems with `self.palette.mode` and thus `self.im` and `self.load()` is required here.

Reply:
If `self.palette.mode` is going out of sync with `self.im.getpalettemode()`, I'm interested in investigating that for its own sake.

Is it possible your code was written before something like #5552, and `load()` is no longer necessary?

Reply:
> Is it possible your code was written before 

Definitely before. 

Anyway, I think `.endswith('A')` is more correct way to check alpha

Reply:
> ImagePalette is only ever created internally

This could eventually change and `endswith('A')` will stay correct in any way.

Reply:
Ok, I've added a commit.